UniversityApp.DataAccess/Repositories/StudentsRepository.cs

```diff
@@ -22,6 +25,27 @@ public async Task<List<Student>> GetStudents()
             return result;
         }
 
+        public async Task<(List<Student> Students, int TotalRecordCount)> GetStudents(List<Expression<Func<Student, bool>>> predicates, int skip, int amount)
```

Comment:
skip and amount should be named in business way -> page/pageNumber. Logic for calculating skip should be here